The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7227559091/job/19695451468) [(plain)](https://github.com/rust-lang/rust/commit/2f1a0315318433ba420e72204acad8a703885871/checks/19695451468/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=Enselic
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_99863706-230e-4b48-bec3-9bc76cd6bfa2
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=fix-IncorrectCguReuseType
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_99863706-230e-4b48-bec3-9bc76cd6bfa2
GITHUB_REF=refs/pull/118973/merge
GITHUB_REF_NAME=118973/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=bc80b894df4ae2038059acf8d1d5d5debd7ea0f8
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_99863706-230e-4b48-bec3-9bc76cd6bfa2
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_99863706-230e-4b48-bec3-9bc76cd6bfa2
GITHUB_TRIGGERING_ACTOR=Enselic
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/118973/merge
GITHUB_WORKFLOW_SHA=bc80b894df4ae2038059acf8d1d5d5debd7ea0f8
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
Built container sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Looks like docker image is the same as before, not uploading
https://ci-caches.rust-lang.org/docker/7ebc15c01a233894034d277c8cce4e949f4e7791f66b4727c8fb6e058a0b8171d6152e1441d677cef0653843ceeee469c097b8699b2bb74249e674f6aa1a8813
sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Fri Dec 15 22:15:07 UTC 2023
  network time: Fri, 15 Dec 2023 22:15:07 GMT
  network time: Fri, 15 Dec 2023 22:15:07 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-missing-tools', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: rust.codegen-units-std := 1
---
Uplifting rustc (stage1 -> stage3)
##[group]Testing stage2 compiletest suite=ui-fulldeps mode=ui (x86_64-unknown-linux-gnu)

running 39 tests
.......F........FFF......F.............
failures:

---- [ui] tests/ui-fulldeps/internal-lints/diagnostics.rs stdout ----
diff of stderr:
diff of stderr:

- error: diagnostics should be created using translatable messages
-   --> $DIR/diagnostics.rs:42:17
+ error: You must mark diagnostic structs with `#[must_use]`
3    |
- LL |         handler.struct_err("untranslatable diagnostic")
-    |                 ^^^^^^^^^^
-    |
-    |
- note: the lint level is defined here
-   --> $DIR/diagnostics.rs:6:9
-    |
- LL | #![deny(rustc::untranslatable_diagnostic)]
-    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ LL | / #[diag(no_crate_example)]
+ LL | | struct DeriveDiagnostic {
+ LL | |     #[primary_span]
+ LL | |     span: Span,
+ LL | | }
12 
- error: diagnostics should be created using translatable messages
-   --> $DIR/diagnostics.rs:62:14
-    |
---
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/internal-lints/diagnostics.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/internal-lints/diagnostics" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/internal-lints/diagnostics/auxiliary" "-Z" "unstable-options"
stdout: none
--- stderr -------------------------------
error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(no_crate_example)]
LL | | struct DeriveDiagnostic {
LL | |     #[primary_span]
LL | |     span: Span,
   | |_^

error: aborting due to 1 previous error
------------------------------------------
------------------------------------------


---- [ui] tests/ui-fulldeps/session-diagnostic/enforce_slug_naming.rs stdout ----
diff of stderr:

+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+ LL | / #[diag(compiletest_example, code = "E0123")]
+ LL | |
+ LL | | struct Hello {}
---
status: exit status: 1
command: CARGO_CRATE_NAME="rustc_dummy" RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/session-diagnostic/enforce_slug_naming.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/enforce_slug_naming" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/enforce_slug_naming/auxiliary"
stdout: none
--- stderr -------------------------------
error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(compiletest_example, code = "E0123")]
LL | / #[diag(compiletest_example, code = "E0123")]
LL | | //~^ ERROR diagnostic slug and crate name do not match
LL | | struct Hello {}

error: diagnostic slug and crate name do not match
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/enforce_slug_naming.rs:23:8
   |
---

---- [ui] tests/ui-fulldeps/session-diagnostic/diagnostic-derive-doc-comment-field.rs stdout ----
diff of stderr:

+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example)]
+ LL | | struct Test {
+ LL | |     #[primary_span]
+ LL | |     span: Span,
+ LL | |
+ LL | | }
+    | |_^
+ 
+ 
1 error[E0277]: the trait bound `NotIntoDiagnosticArg: IntoDiagnosticArg` is not satisfied
3    |

25 note: required by a bound in `Diagnostic::set_arg`
25 note: required by a bound in `Diagnostic::set_arg`
26   --> $COMPILER_DIR/rustc_errors/src/diagnostic.rs:LL:CC
- error: aborting due to 2 previous errors
+ error: aborting due to 3 previous errors
29 
30 For more information about this error, try `rustc --explain E0277`.
---
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive-doc-comment-field.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/diagnostic-derive-doc-comment-field" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/diagnostic-derive-doc-comment-field/auxiliary"
stdout: none
--- stderr -------------------------------
error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(no_crate_example)]
LL | | struct Test {
LL | |     #[primary_span]
LL | |     span: Span,
...  |
LL | |     //~^ ERROR the trait bound `NotIntoDiagnosticArg: IntoDiagnosticArg` is not satisfied
   | |_^


error[E0277]: the trait bound `NotIntoDiagnosticArg: IntoDiagnosticArg` is not satisfied
   |
LL | #[derive(Diagnostic)]
   |          ---------- required by a bound introduced by this call
...
---
             i64
             i128
             usize
           and 46 others
note: required by a bound in `DiagnosticBuilder::<'a, G>::set_arg`
  --> /rustc/FAKE_PREFIX/compiler/rustc_errors/src/diagnostic_builder.rs:619:5
   = note: this error originates in the macro `forward` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NotIntoDiagnosticArg: IntoDiagnosticArg` is not satisfied
   |
LL | #[derive(Subdiagnostic)]
   |          ------------- required by a bound introduced by this call
...
---
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/session-diagnostic/invalid-variable.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/invalid-variable/a" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/invalid-variable/auxiliary" "--test"
stdout: none
--- stderr -------------------------------
error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(no_crate_bad_reference)]
LL | | struct BadRef;

error: aborting due to 1 previous error
------------------------------------------



---- [ui] tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs stdout ----
diff of stderr:

+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct Hello {}
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct HelloWarn {}
+ 
1 error: unsupported type attribute for diagnostic derive enum
2   --> $DIR/diagnostic-derive.rs:43:1
3    |
3    |

20    |
21    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
22 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | #[diag = "E0123"]
+ LL | |
+ LL | | struct WrongStructAttrStyle {}
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[nonsense(no_crate_example, code = "E0123")]
+ LL | |
+ LL | |
+ LL | |
+ LL | | struct InvalidStructAttr {}
+ 
+ 
23 error: `#[nonsense(...)]` is not a valid attribute
25    |

38    |
38    |
39    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
40 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+ LL | / #[diag("E0123")]
+ LL | |
+ LL | |
+ LL | | struct InvalidLitNestedAttr {}
+ 
41 error: diagnostic slug not specified
42   --> $DIR/diagnostic-derive.rs:66:1
43    |
43    |

48    |
49    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
50 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(nonsense, code = "E0123")]
+ LL | |
+ LL | | struct InvalidNestedStructAttr {}
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(nonsense("foo"), code = "E0123", slug = "foo")]
+ LL | |
+ LL | |
+ LL | | struct InvalidNestedStructAttr1 {}
+ 
51 error: diagnostic slug must be the first argument
52   --> $DIR/diagnostic-derive.rs:76:16
53    |
53    |

65    |
66    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
67 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(nonsense = "...", code = "E0123", slug = "foo")]
+ LL | |
+ LL | |
+ LL | | struct InvalidNestedStructAttr2 {}
+ 
68 error: unknown argument
69   --> $DIR/diagnostic-derive.rs:82:8
70    |
70    |

84    |
85    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
86 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(nonsense = 4, code = "E0123", slug = "foo")]
+ LL | |
+ LL | |
+ LL | | struct InvalidNestedStructAttr3 {}
+ 
87 error: unknown argument
88   --> $DIR/diagnostic-derive.rs:88:8
89    |
89    |

103    |
104    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
105 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123", slug = "foo")]
+ LL | |
+ LL | | struct InvalidNestedStructAttr4 {}
+ 
106 error: unknown argument
107   --> $DIR/diagnostic-derive.rs:94:42
108    |
108    |

111    |
112    = note: only the `code` parameter is valid after the slug
113 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct WrongPlaceField {
+ LL | |     #[suggestion = "bar"]
+ LL | |
+ LL | |     sp: Span,
+ LL | | }
+ 
+ 
114 error: `#[suggestion = ...]` is not a valid attribute
116    |


117 LL |     #[suggestion = "bar"]
119 
119 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | #[diag(no_crate_example, code = "E0456")]
+ LL | |
+ LL | |
+ LL | | struct DiagSpecifiedTwice {}
+ 
120 error: specified multiple times
121   --> $DIR/diagnostic-derive.rs:108:8
122    |
122    |

141 LL | #[diag(no_crate_example, code = "E0123")]
142    |                          ^^^^
143 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0456", code = "E0457")]
+ LL | |
+ LL | | struct CodeSpecifiedTwice {}
+ 
144 error: specified multiple times
145   --> $DIR/diagnostic-derive.rs:114:42
146    |
146    |

153 LL | #[diag(no_crate_example, code = "E0456", code = "E0457")]
155 
155 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, no_crate::example, code = "E0456")]
+ LL | |
+ LL | | struct SlugSpecifiedTwice {}
+ 
156 error: diagnostic slug must be the first argument
157   --> $DIR/diagnostic-derive.rs:119:43
158    |
158    |

159 LL | #[diag(no_crate_example, no_crate::example, code = "E0456")]
160    |                                           ^
161 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | struct KindNotProvided {}
+ 
162 error: diagnostic slug not specified
163   --> $DIR/diagnostic-derive.rs:124:1
164    |
164    |

167    |
168    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
169 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+ LL | / #[diag(code = "E0456")]
+ LL | |
+ LL | |
+ LL | | struct SlugNotProvided {}
+ 
170 error: diagnostic slug not specified
171   --> $DIR/diagnostic-derive.rs:127:1
172    |
172    |

177    |
178    = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
179 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example)]
+ LL | | struct CodeNotProvided {}
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct MessageWrongType {
+ LL | |     #[primary_span]
+ LL | |
+ LL | |     foo: String,
+ LL | | }
+ 
+ 
180 error: the `#[primary_span]` attribute can only be applied to fields of type `Span` or `MultiSpan`
182    |


183 LL |     #[primary_span]
185 
185 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct InvalidPathFieldAttr {
+ LL | |     #[nonsense]
+ LL | |
+ LL | |
+ LL | |     foo: String,
+ LL | | }
+ 
+ 
186 error: `#[nonsense]` is not a valid attribute
188    |


189 LL |     #[nonsense]
191 
191 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct ErrorWithField {
+ LL | |     name: String,
+ LL | |     #[label(no_crate_label)]
+ LL | |     span: Span,
+ LL | | }
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct ErrorWithMessageAppliedToField {
+ LL | |     #[label(no_crate_label)]
+ LL | |
+ LL | |     name: String,
+ LL | | }
+ 
+ 
192 error: the `#[label(...)]` attribute can only be applied to fields of type `Span` or `MultiSpan`
194    |


195 LL |     #[label(no_crate_label)]
197 
197 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct ErrorWithNonexistentField {
+ LL | |     #[suggestion(no_crate_suggestion, code = "{name}")]
+ LL | |
+ LL | |     suggestion: (Span, Applicability),
+ LL | | }
+ 
198 error: `name` doesn't refer to a field on this type
199   --> $DIR/diagnostic-derive.rs:171:46
200    |
200    |

201 LL |     #[suggestion(no_crate_suggestion, code = "{name}")]
203 
203 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct ErrorMissingClosingBrace {
+ LL | |     #[suggestion(no_crate_suggestion, code = "{name")]
+ LL | |     suggestion: (Span, Applicability),
+ LL | |     name: String,
+ LL | |     val: usize,
+ LL | | }
+ 
+ 
204 error: invalid format string: expected `'}'` but string was terminated
206    |


210    = note: if you intended to print `{`, you can escape it using `{{`
211    = note: this error originates in the derive macro `Diagnostic` (in Nightly builds, run with -Z macro-backtrace for more info)
212 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct ErrorMissingOpeningBrace {
+ LL | |     #[suggestion(no_crate_suggestion, code = "name}")]
+ LL | |     suggestion: (Span, Applicability),
+ LL | |     name: String,
+ LL | |     val: usize,
+ LL | | }
+ 
+ 
213 error: invalid format string: unmatched `}` found
215    |


219    = note: if you intended to print `}`, you can escape it using `}}`
220    = note: this error originates in the derive macro `Diagnostic` (in Nightly builds, run with -Z macro-backtrace for more info)
221 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct LabelOnSpan {
+ LL | |     #[label(no_crate_label)]
+ LL | |     sp: Span,
+ LL | | }
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct LabelOnNonSpan {
+ LL | |     #[label(no_crate_label)]
+ LL | |
+ LL | |     id: u32,
+ LL | | }
+ 
+ 
222 error: the `#[label(...)]` attribute can only be applied to fields of type `Span` or `MultiSpan`
224    |


225 LL |     #[label(no_crate_label)]
227 
227 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct Suggest {
+ LL | |     #[suggestion(no_crate_suggestion, code = "This is the suggested code")]
+ LL | |     #[suggestion(no_crate_suggestion, code = "This is the suggested code", style = "normal")]
+ ...  |
+ LL | |     suggestion: (Span, Applicability),
+ LL | | }
+ 
Build completed unsuccessfully in 0:13:57
Build completed unsuccessfully in 0:13:57
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct SuggestWithoutCode {
+ LL | |     #[suggestion(no_crate_suggestion)]
+ LL | |
+ LL | |     suggestion: (Span, Applicability),
+ LL | | }
+ 
228 error: suggestion without `code = "..."`
229   --> $DIR/diagnostic-derive.rs:225:5
230    |
230    |

231 LL |     #[suggestion(no_crate_suggestion)]
233 
233 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct SuggestWithBadKey {
+ LL | |     #[suggestion(nonsense = "bar")]
+ LL | |
+ LL | |
+ LL | |     suggestion: (Span, Applicability),
+ LL | | }
+ 
234 error: invalid nested attribute
235   --> $DIR/diagnostic-derive.rs:233:18
236    |
236    |

245 LL |     #[suggestion(nonsense = "bar")]
247 
247 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct SuggestWithShorthandMsg {
+ LL | |     #[suggestion(msg = "bar")]
+ LL | |
+ LL | |
+ LL | |     suggestion: (Span, Applicability),
+ LL | | }
+ 
248 error: invalid nested attribute
249   --> $DIR/diagnostic-derive.rs:242:18
250    |
250    |

259 LL |     #[suggestion(msg = "bar")]
261 
261 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct SuggestWithoutMsg {
+ LL | |     #[suggestion(code = "bar")]
+ LL | |     suggestion: (Span, Applicability),
+ LL | | }
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct SuggestWithTypesSwapped {
+ LL | |     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
+ LL | |     suggestion: (Applicability, Span),
+ LL | | }
+ 
+ 
+ error: You must mark diagnostic structs with `#[must_use]`
+    |
+    |
+ LL | / #[diag(no_crate_example, code = "E0123")]
+ LL | | struct SuggestWithWrongTypeApplicabilityOnly {
+ LL | |     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
+ LL | |
+ LL | |     suggestion: Applicability,
+ LL | | }
+ 
---
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/diagnostic-derive" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/session-diagnostic/diagnostic-derive/auxiliary"
stdout: none
--- stderr -------------------------------
error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct Hello {}
   | |_______________^
   | |_______________^

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct HelloWarn {}

error: unsupported type attribute for diagnostic derive enum
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:43:1
   |
---
   |
LL |     Foo,
   |     ^^^
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`
error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:48:5
   |
LL |     Bar,
LL |     Bar,
   |     ^^^
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | #[diag = "E0123"]
LL | | #[diag = "E0123"]
LL | | //~^ ERROR failed to resolve: maybe a missing crate `core`
LL | | struct WrongStructAttrStyle {}


error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[nonsense(no_crate_example, code = "E0123")]
LL | | //~^ ERROR `#[nonsense(...)]` is not a valid attribute
LL | | //~^^ ERROR diagnostic slug not specified
LL | | //~^^^ ERROR cannot find attribute `nonsense` in this scope
LL | | struct InvalidStructAttr {}


error: `#[nonsense(...)]` is not a valid attribute
   |
   |
LL | #[nonsense(no_crate_example, code = "E0123")]

error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:59:1
   |
   |
LL | / #[nonsense(no_crate_example, code = "E0123")]
LL | | //~^ ERROR `#[nonsense(...)]` is not a valid attribute
LL | | //~^^ ERROR diagnostic slug not specified
LL | | //~^^^ ERROR cannot find attribute `nonsense` in this scope
LL | | struct InvalidStructAttr {}
   |
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag("E0123")]
LL | / #[diag("E0123")]
LL | | //~^ ERROR diagnostic slug not specified
LL | | struct InvalidLitNestedAttr {}

error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:66:1
   |
   |
LL | / #[diag("E0123")]
LL | | //~^ ERROR diagnostic slug not specified
LL | | struct InvalidLitNestedAttr {}
   |
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(nonsense, code = "E0123")]
LL | | //~^ ERROR cannot find value `nonsense` in module `crate::fluent_generated`
LL | | struct InvalidNestedStructAttr {}


error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(nonsense("foo"), code = "E0123", slug = "foo")]
LL | | //~^ ERROR diagnostic slug must be the first argument
LL | | //~| ERROR diagnostic slug not specified
LL | | struct InvalidNestedStructAttr1 {}

error: diagnostic slug must be the first argument
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:76:16
   |
   |
LL | #[diag(nonsense("foo"), code = "E0123", slug = "foo")]

error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:76:1
   |
   |
LL | / #[diag(nonsense("foo"), code = "E0123", slug = "foo")]
LL | | //~^ ERROR diagnostic slug must be the first argument
LL | | //~| ERROR diagnostic slug not specified
LL | | struct InvalidNestedStructAttr1 {}
   |
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(nonsense = "...", code = "E0123", slug = "foo")]
LL | | //~^ ERROR unknown argument
LL | | //~| ERROR diagnostic slug not specified
LL | | struct InvalidNestedStructAttr2 {}

error: unknown argument
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:82:8
   |
   |
LL | #[diag(nonsense = "...", code = "E0123", slug = "foo")]
   |
   |
   = note: only the `code` parameter is valid after the slug
error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:82:1
   |
   |
LL | / #[diag(nonsense = "...", code = "E0123", slug = "foo")]
LL | | //~^ ERROR unknown argument
LL | | //~| ERROR diagnostic slug not specified
LL | | struct InvalidNestedStructAttr2 {}
   |
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(nonsense = 4, code = "E0123", slug = "foo")]
LL | | //~^ ERROR unknown argument
LL | | //~| ERROR diagnostic slug not specified
LL | | struct InvalidNestedStructAttr3 {}

error: unknown argument
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:88:8
   |
   |
LL | #[diag(nonsense = 4, code = "E0123", slug = "foo")]
   |
   |
   = note: only the `code` parameter is valid after the slug
error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:88:1
   |
   |
LL | / #[diag(nonsense = 4, code = "E0123", slug = "foo")]
LL | | //~^ ERROR unknown argument
LL | | //~| ERROR diagnostic slug not specified
LL | | struct InvalidNestedStructAttr3 {}
   |
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(no_crate_example, code = "E0123", slug = "foo")]
LL | | //~^ ERROR unknown argument
LL | | struct InvalidNestedStructAttr4 {}

error: unknown argument
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:94:42
   |
   |
LL | #[diag(no_crate_example, code = "E0123", slug = "foo")]
   |
   |
   = note: only the `code` parameter is valid after the slug

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct WrongPlaceField {
LL | |     #[suggestion = "bar"]
LL | |     //~^ ERROR `#[suggestion = ...]` is not a valid attribute
LL | |     sp: Span,
   | |_^


error: `#[suggestion = ...]` is not a valid attribute
   |
   |
LL |     #[suggestion = "bar"]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | #[diag(no_crate_example, code = "E0456")]
LL | | #[diag(no_crate_example, code = "E0456")]
LL | | //~^ ERROR specified multiple times
LL | | //~^^ ERROR specified multiple times
LL | | struct DiagSpecifiedTwice {}

error: specified multiple times
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:108:8
   |
---
   |
LL | #[diag(no_crate_example, code = "E0123")]
   |                          ^^^^

error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(no_crate_example, code = "E0456", code = "E0457")]
LL | | //~^ ERROR specified multiple times
LL | | struct CodeSpecifiedTwice {}

error: specified multiple times
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:114:42
   |
   |
LL | #[diag(no_crate_example, code = "E0456", code = "E0457")]
   |
note: previously specified here
  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:114:26
   |
   |
LL | #[diag(no_crate_example, code = "E0456", code = "E0457")]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, no_crate::example, code = "E0456")]
LL | / #[diag(no_crate_example, no_crate::example, code = "E0456")]
LL | | //~^ ERROR diagnostic slug must be the first argument
LL | | struct SlugSpecifiedTwice {}

error: diagnostic slug must be the first argument
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:119:43
   |
   |
LL | #[diag(no_crate_example, no_crate::example, code = "E0456")]
   |                                           ^

error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | struct KindNotProvided {} //~ ERROR diagnostic slug not specified

error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:124:1
   |
   |
LL | struct KindNotProvided {} //~ ERROR diagnostic slug not specified
   |
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(code = "E0456")]
LL | / #[diag(code = "E0456")]
LL | | //~^ ERROR diagnostic slug not specified
LL | | struct SlugNotProvided {}

error: diagnostic slug not specified
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:127:1
   |
   |
LL | / #[diag(code = "E0456")]
LL | | //~^ ERROR diagnostic slug not specified
LL | | struct SlugNotProvided {}
   |
   |
   = help: specify the slug as the first argument to the `#[diag(...)]` attribute, such as `#[diag(hir_analysis_example_error)]`

error: You must mark diagnostic structs with `#[must_use]`
   |
   |
LL | / #[diag(no_crate_example)]
LL | | struct CodeNotProvided {}


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct MessageWrongType {
LL | |     #[primary_span]
LL | |     //~^ ERROR `#[primary_span]` attribute can only be applied to fields of type `Span` or `MultiSpan`
LL | |     foo: String,
   | |_^


error: the `#[primary_span]` attribute can only be applied to fields of type `Span` or `MultiSpan`
   |
   |
LL |     #[primary_span]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct InvalidPathFieldAttr {
LL | |     #[nonsense]
LL | |     //~^ ERROR `#[nonsense]` is not a valid attribute
LL | |     //~^^ ERROR cannot find attribute `nonsense` in this scope
LL | |     foo: String,
   | |_^


error: `#[nonsense]` is not a valid attribute
   |
   |
LL |     #[nonsense]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorWithField {
LL | |     name: String,
LL | |     name: String,
LL | |     #[label(no_crate_label)]
LL | |     span: Span,
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorWithMessageAppliedToField {
LL | |     #[label(no_crate_label)]
LL | |     //~^ ERROR the `#[label(...)]` attribute can only be applied to fields of type `Span` or `MultiSpan`
LL | | }
   | |_^


error: the `#[label(...)]` attribute can only be applied to fields of type `Span` or `MultiSpan`
   |
   |
LL |     #[label(no_crate_label)]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorWithNonexistentField {
LL | |     #[suggestion(no_crate_suggestion, code = "{name}")]
LL | |     //~^ ERROR `name` doesn't refer to a field on this type
LL | |     suggestion: (Span, Applicability),
   | |_^

error: `name` doesn't refer to a field on this type
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:171:46
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:171:46
   |
LL |     #[suggestion(no_crate_suggestion, code = "{name}")]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorMissingClosingBrace {
LL | |     #[suggestion(no_crate_suggestion, code = "{name")]
LL | |     suggestion: (Span, Applicability),
LL | |     val: usize,
LL | | }
   | |_^


error: invalid format string: expected `'}'` but string was terminated
   |
LL | #[derive(Diagnostic)]
LL | #[derive(Diagnostic)]
   |          ^^^^^^^^^^ expected `'}'` in format string
   |
   = note: if you intended to print `{`, you can escape it using `{{`
   = note: this error originates in the derive macro `Diagnostic` (in Nightly builds, run with -Z macro-backtrace for more info)

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorMissingOpeningBrace {
LL | | struct ErrorMissingOpeningBrace {
LL | |     #[suggestion(no_crate_suggestion, code = "name}")]
LL | |     suggestion: (Span, Applicability),
LL | |     val: usize,
LL | | }
   | |_^


error: invalid format string: unmatched `}` found
   |
LL | #[derive(Diagnostic)]
LL | #[derive(Diagnostic)]
   |          ^^^^^^^^^^ unmatched `}` in format string
   |
   = note: if you intended to print `}`, you can escape it using `}}`
   = note: this error originates in the derive macro `Diagnostic` (in Nightly builds, run with -Z macro-backtrace for more info)

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct LabelOnSpan {
LL | | struct LabelOnSpan {
LL | |     #[label(no_crate_label)]
LL | |     sp: Span,
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct LabelOnNonSpan {
LL | |     #[label(no_crate_label)]
LL | |     //~^ ERROR the `#[label(...)]` attribute can only be applied to fields of type `Span` or `MultiSpan`
LL | |     id: u32,
   | |_^


error: the `#[label(...)]` attribute can only be applied to fields of type `Span` or `MultiSpan`
   |
   |
LL |     #[label(no_crate_label)]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct Suggest {
LL | | struct Suggest {
LL | |     #[suggestion(no_crate_suggestion, code = "This is the suggested code")]
LL | |     #[suggestion(no_crate_suggestion, code = "This is the suggested code", style = "normal")]
...  |
LL | |     suggestion: (Span, Applicability),
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithoutCode {
LL | |     #[suggestion(no_crate_suggestion)]
LL | |     //~^ ERROR suggestion without `code = "..."`
LL | |     suggestion: (Span, Applicability),
   | |_^

error: suggestion without `code = "..."`
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:225:5
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:225:5
   |
LL |     #[suggestion(no_crate_suggestion)]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithBadKey {
LL | |     #[suggestion(nonsense = "bar")]
LL | |     //~^ ERROR invalid nested attribute
LL | |     //~| ERROR suggestion without `code = "..."`
LL | |     suggestion: (Span, Applicability),
   | |_^

error: invalid nested attribute
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:233:18
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:233:18
   |
LL |     #[suggestion(nonsense = "bar")]
   |
   |
   = help: only `no_span`, `style`, `code` and `applicability` are valid nested attributes
error: suggestion without `code = "..."`
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:233:5
   |
   |
LL |     #[suggestion(nonsense = "bar")]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithShorthandMsg {
LL | |     #[suggestion(msg = "bar")]
LL | |     //~^ ERROR invalid nested attribute
LL | |     //~| ERROR suggestion without `code = "..."`
LL | |     suggestion: (Span, Applicability),
   | |_^

error: invalid nested attribute
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:242:18
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:242:18
   |
LL |     #[suggestion(msg = "bar")]
   |
   |
   = help: only `no_span`, `style`, `code` and `applicability` are valid nested attributes
error: suggestion without `code = "..."`
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:242:5
   |
   |
LL |     #[suggestion(msg = "bar")]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithoutMsg {
LL | |     #[suggestion(code = "bar")]
LL | |     suggestion: (Span, Applicability),
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithTypesSwapped {
LL | |     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
LL | |     suggestion: (Applicability, Span),
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithWrongTypeApplicabilityOnly {
LL | |     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
LL | |     //~^ ERROR wrong field type for suggestion
LL | |     suggestion: Applicability,
   | |_^

error: wrong field type for suggestion
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:265:5
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:265:5
   |
LL | /     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
LL | |     //~^ ERROR wrong field type for suggestion
LL | |     suggestion: Applicability,
   |
   |
   = help: `#[suggestion(...)]` should be applied to fields of type `Span` or `(Span, Applicability)`

error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithSpanOnly {
LL | |     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
LL | |     suggestion: Span,
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithDuplicateSpanAndApplicability {
LL | |     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
LL | |     suggestion: (Span, Span, Applicability),
LL | |     //~^ ERROR specified multiple times
   | |_^

error: specified multiple times
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:281:24
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:281:24
   |
LL |     suggestion: (Span, Span, Applicability),
   |
note: previously specified here
  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:281:18
   |
   |
LL |     suggestion: (Span, Span, Applicability),


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct SuggestWithDuplicateApplicabilityAndSpan {
LL | |     #[suggestion(no_crate_suggestion, code = "This is suggested code")]
LL | |     suggestion: (Applicability, Applicability, Span),
LL | |     //~^ ERROR specified multiple times
   | |_^

error: specified multiple times
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:289:33
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:289:33
   |
LL |     suggestion: (Applicability, Applicability, Span),
   |
note: previously specified here
  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:289:18
   |
   |
LL |     suggestion: (Applicability, Applicability, Span),


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct WrongKindOfAnnotation {
LL | |     #[label = "bar"]
LL | |     //~^ ERROR `#[label = ...]` is not a valid attribute
LL | |     z: Span,
   | |_^


error: `#[label = ...]` is not a valid attribute
   |
   |
LL |     #[label = "bar"]


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct OptionsInErrors {
LL | | struct OptionsInErrors {
LL | |     #[label(no_crate_label)]
LL | |     label: Option<Span>,
LL | |     #[suggestion(no_crate_suggestion, code = "...")]
LL | |     opt_sugg: Option<(Span, Applicability)>,
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0456")]
LL | | struct MoveOutOfBorrowError<'tcx> {
LL | |     name: Ident,
LL | |     name: Ident,
LL | |     ty: Ty<'tcx>,
...  |
LL | |     opt_sugg: Option<(Span, Applicability)>,
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorWithLifetime<'a> {
LL | |     #[label(no_crate_label)]
LL | |     span: Span,
LL | |     name: &'a str,
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorWithDefaultLabelAttr<'a> {
LL | |     #[label]
LL | |     span: Span,
LL | |     name: &'a str,
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ArgFieldWithoutSkip {
LL | |     #[primary_span]
LL | |     span: Span,
LL | |     other: Hello,
LL | |     //~^ ERROR the trait bound `Hello: IntoDiagnosticArg` is not satisfied
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ArgFieldWithSkip {
LL | |     #[primary_span]
LL | |     span: Span,
LL | |     other: Hello,
LL | | }
   | |_^


error: You must mark diagnostic structs with `#[must_use]`
   |
LL | / #[diag(no_crate_example, code = "E0123")]
LL | / #[diag(no_crate_example, code = "E0123")]
LL | | struct ErrorWithSpannedNote {
LL | |     #[note]
LL | |     span: Span,
   | |_^

---

error: cannot find attribute `multipart_suggestion` in this scope
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:648:7
   |
LL |     #[multipart_suggestion(no_crate_suggestion)]

error[E0425]: cannot find value `nonsense` in module `crate::fluent_generated`
##[error]  --> /checkout/tests/ui-fulldeps/session-diagnostic/diagnostic-derive.rs:71:8
   |
   |
LL | #[diag(nonsense, code = "E0123")]
   |        ^^^^^^^^ not found in `crate::fluent_generated`

error[E0425]: cannot find value `__code_34` in this scope
   |
   |
LL | #[derive(Diagnostic)] //~ ERROR cannot find value `__code_34` in this scope
   |
   = note: this error originates in the derive macro `Diagnostic` (in Nightly builds, run with -Z macro-backtrace for more info)


error[E0277]: the trait bound `Hello: IntoDiagnosticArg` is not satisfied
   |
LL | #[derive(Diagnostic)]
   |          ---------- required by a bound introduced by this call
...
...
LL |     other: Hello,
   |            ^^^^^ the trait `IntoDiagnosticArg` is not implemented for `Hello`
   = help: the following other types implement trait `IntoDiagnosticArg`:
             bool
             char
             i8
             i8
             i16
             i32
             i64
             i128
             usize
           and 65 others
note: required by a bound in `DiagnosticBuilder::<'a, G>::set_arg`
  --> /rustc/FAKE_PREFIX/compiler/rustc_errors/src/diagnostic_builder.rs:619:5
   = note: this error originates in the macro `forward` (in Nightly builds, run with -Z macro-backtrace for more info)
error: aborting due to 178 previous errors

Some errors have detailed explanations: E0277, E0425, E0433.
For more information about an error, try `rustc --explain E0277`.
```

</details>

--- a/compiler/rustc_codegen_ssa/src/assert_module_sources.rs
+++ b/compiler/rustc_codegen_ssa/src/assert_module_sources.rs
@@ -278,13 +278,13 @@ impl CguReuseTracker {
 
                     if error {
                         let at_least = if at_least { 1 } else { 0 };
-                        errors::IncorrectCguReuseType {
+                        sess.emit_err(errors::IncorrectCguReuseType {
                             span: *error_span,
                             cgu_user_name,
                             actual_reuse,
                             expected_reuse,
                             at_least,
-                        };
+                        });
                     }
                 } else {
                     sess.emit_fatal(errors::CguNotRecorded { cgu_user_name, cgu_name });

--- a/tests/incremental/thinlto/cgu_invalidated_via_import.rs
+++ b/tests/incremental/thinlto/cgu_invalidated_via_import.rs
@@ -14,14 +14,14 @@
                             kind="no")]
 #![rustc_expected_cgu_reuse(module="cgu_invalidated_via_import-foo",
                             cfg="cfail3",
-                            kind="post-lto")]
+                            kind="pre-lto")] // Should be "post-lto", see issue #119076
 
 #![rustc_expected_cgu_reuse(module="cgu_invalidated_via_import-bar",
                             cfg="cfail2",
                             kind="pre-lto")]
 #![rustc_expected_cgu_reuse(module="cgu_invalidated_via_import-bar",
                             cfg="cfail3",
-                            kind="post-lto")]
+                            kind="pre-lto")] // Should be "post-lto", see issue #119076
 
 mod foo {
 

--- a/tests/incremental/thinlto/cgu_keeps_identical_fn.rs
+++ b/tests/incremental/thinlto/cgu_keeps_identical_fn.rs
@@ -9,21 +9,25 @@
 
 #![feature(rustc_attrs)]
 #![crate_type = "rlib"]
-#![rustc_expected_cgu_reuse(module = "cgu_keeps_identical_fn-foo", cfg = "cfail2", kind = "no")]
+#![rustc_expected_cgu_reuse(
+    module = "cgu_keeps_identical_fn-foo",
+    cfg = "cfail2",
+    kind = "pre-lto"
+)]
 #![rustc_expected_cgu_reuse(
     module = "cgu_keeps_identical_fn-foo",
     cfg = "cfail3",
-    kind = "post-lto"
+    kind = "pre-lto" // Should be "post-lto", see issue #119076
 )]
 #![rustc_expected_cgu_reuse(
     module = "cgu_keeps_identical_fn-bar",
     cfg = "cfail2",
-    kind = "post-lto"
+    kind = "pre-lto" // Should be "post-lto", see issue #119076
 )]
 #![rustc_expected_cgu_reuse(
     module = "cgu_keeps_identical_fn-bar",
     cfg = "cfail3",
-    kind = "post-lto"
+    kind = "pre-lto" // Should be "post-lto", see issue #119076
 )]
 
 mod foo {

--- a/tests/incremental/thinlto/independent_cgus_dont_affect_each_other.rs
+++ b/tests/incremental/thinlto/independent_cgus_dont_affect_each_other.rs
@@ -13,21 +13,21 @@
                             kind="no")]
 #![rustc_expected_cgu_reuse(module="independent_cgus_dont_affect_each_other-foo",
                             cfg="cfail3",
-                            kind="post-lto")]
+                            kind="pre-lto")] // Should be "post-lto", see issue #119076
 
 #![rustc_expected_cgu_reuse(module="independent_cgus_dont_affect_each_other-bar",
                             cfg="cfail2",
                             kind="pre-lto")]
 #![rustc_expected_cgu_reuse(module="independent_cgus_dont_affect_each_other-bar",
                             cfg="cfail3",
-                            kind="post-lto")]
+                            kind="pre-lto")] // Should be "post-lto", see issue #119076
 
 #![rustc_expected_cgu_reuse(module="independent_cgus_dont_affect_each_other-baz",
                             cfg="cfail2",
-                            kind="post-lto")]
+                            kind="pre-lto")] // Should be "post-lto", see issue #119076
 #![rustc_expected_cgu_reuse(module="independent_cgus_dont_affect_each_other-baz",
                             cfg="cfail3",
-                            kind="post-lto")]
+                            kind="pre-lto")] // Should be "post-lto", see issue #119076
 mod foo {
 
     #[cfg(cfail1)]

